engine/runtime/src/main/java/org/enso/interpreter/node/callable/IndirectInvokeCallableNode.java

```diff
@@ -88,7 +89,7 @@ Object invokeWithWarnings(
               isTail);
 
       Warning[] extracted = warnings.getWarnings(warning, null, false);
```

Comment:
If we want to keep decent performance, than we have to stop playing with `Warning[]` arrays. Extracting warnings should be _single instruction operation_ - just **obtain a reference** to `EnsoHashMap`.

The same applies to _appending_ - it should use the **same reference** to `EnsoHashMap` and associate it with the `result`. It should perform _no array manipulation_ at all.

Reply:
As of 01381e6cc223801e389f37d6bd65ede531475e3d, the solution you propose here is already implemented. Moreover, *reassignments* were already removed altogether - we no longer keep track of the reassignments in the warnings. Resolving this conversation as outdated.